### PR TITLE
Implement logic for transition banner.

### PR DESCRIPTION
**Pull request by mrstegeman**
References #2612

### CODE REVIEW

File: src/ssltunnel.js [hunk 1]
```diff
@@ -77,10 +77,17 @@ const TunnelService = {
   start: function(response, urlredirect) {
     Settings.get('tunneltoken').then((result) => {
       if (typeof result === 'object') {
+        if (!result.base) {
+          // handle legacy tunnels
+          result.base = 'mozilla-iot.org';
```
Comment (benfrancis) [inline src/ssltunnel.js:82]: Should this be set as a constant somewhere?

Comment (benfrancis) [inline src/ssltunnel.js:82]: Should this be set as a constant somewhere?

File: static/index.html [hunk 2]
```diff
@@ -625,7 +627,7 @@ <h3 class="logs-header" data-l10n-id="logs"></h3>
       <div id="transition-dialog-step-1">
         <img id="transition-dialog-step-1-wordmark" src="/images/webthings-lockup.svg"
              data-l10n-id="transition-dialog-wordmark" />
-        <button id="transition-dialog-close" class="close-button"></button>
+        <button id="transition-dialog-step-1-close" class="close-button"></button>
```
Comment (benfrancis) [inline static/index.html:630]: Sorry I spotted this duplicate ID before and intended to fix it. This will require a CSS fix to update the selectors.

**Review by benfrancis — APPROVED** [review]
Looks good. A couple of minor comments.

File: static/js/views/transition-dialog.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      this.elements.banner.main.classList.add('hidden');
+
+      API.putJson('/settings/transition', {skipped: true}).catch((e) => {
+        console.error('Failed to skip transition:', e);
+      });
+    });
+
+    this.elements.banner.choose.addEventListener('click', () => {
+      this.elements.banner.main.classList.add('hidden');
+      this.elements.dialog.main.classList.remove('hidden');
```
Comment (benfrancis) [inline static/js/views/transition-dialog.js:57]: Oh one other thing, we'd ideally show a full screen scrim at this point as well.

File: static/js/views/transition-dialog.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    // Close buttons
+    this.elements.banner.close.addEventListener('click', () => {
+      this.elements.banner.main.classList.add('hidden');
+    });
+    this.elements.dialog.step1.close.addEventListener('click', () => {
+      this.elements.dialog.main.classList.add('hidden');
+    });
+    this.elements.dialog.step2.close.addEventListener('click', () => {
+      this.elements.dialog.main.classList.add('hidden');
```
Comment (benfrancis) [inline static/js/views/transition-dialog.js:43]: ...and hide the scrim here

File: static/index.html (hunk 2 above)
Comment (mrstegeman) [reply static/index.html:630]: I'll keep that in mind.

File: src/ssltunnel.js (hunk 1 above)
Comment (mrstegeman) [reply src/ssltunnel.js:82]: Potentially. This code will probably be removed pretty quickly, but I may need to add a constant later to handle the rest of the transition logic.

File: static/js/views/transition-dialog.js (hunk 3 above)
Comment (mrstegeman) [reply static/js/views/transition-dialog.js:57]: I'll do that in the next PR.

File: src/ssltunnel.js (hunk 1 above)
Comment (benfrancis) [reply src/ssltunnel.js:82]: Yeah I'm looking forward to ripping all this transition code out again as soon as possible!

File: static/index.html (hunk 2 above)
Comment (mrstegeman) [reply static/index.html:630]: Fixed in 599dae9908aa1e8158c5344700899f7b8d4c5c3d

File: static/js/views/transition-dialog.js (hunk 3 above)
Comment (mrstegeman) [reply static/js/views/transition-dialog.js:57]: See #2654

**Pull request merged**